### PR TITLE
Add Authentication hook to frontend and backend

### PR DESCRIPTION
**Pull request by KnightNiwrem**

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ const Router = new VueRouter({
 });
 
 Router.beforeEach((to, from, next) => {
-  if (to.matched.some(record => record.meta.requiresAuth) && !store.state.isLoggedIn) {
+  window.store = store;
```
Comment (ngsikai) [inline src/router/index.js]: Why are we assigning this to global?

File: src/store/login/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          } else {
+            reject('User cancelled login authentication');
+          }
+        });
+      }).then((authResponse) => {
+        return dispatch('serverAuthentication', {
+          authResponse
+        });
+      }).then((serverResponse) => {
+        commit('login', serverResponse);
```
Comment (ngsikai) [inline src/store/login/index.js:39]: It has been a convention to have a separate string mapping for other devs to see at a glance what mutations are available for use. Do you think we won't need that?

File: src/router/index.js (hunk 1 above)
Comment (KnightNiwrem) [reply src/router/index.js]: Oh. For debugging. Will remove

File: src/store/login/index.js (hunk 2 above)
Comment (KnightNiwrem) [reply src/store/login/index.js:39]: KIV

**Pull request merged**